diagnostic/engine.go

```diff
@@ -144,112 +236,6 @@ func (c *conflict) addSimilarConflict(conflict conflict) {
 	c.similarConflicts = append(c.similarConflicts, &conflict)
 }
 
-// ConflictList stores a list of conflicts.
-type ConflictList struct {
-	conflicts  []conflict
-	NoGrouping bool // if set to true, conflicts are not grouped by nil path. (Example use case: no-infer unit tests)
-}
-
-// AddSingleAssertionConflict adds a new single assertion conflict to the list of conflicts.
-func (l *ConflictList) AddSingleAssertionConflict(pass *analysis.Pass, trigger annotation.FullTrigger) {
-	t := fullTriggerAsPrimitive(pass, trigger)
-	c := conflict{
-		pos:  t.Pos,
-		flow: nilFlow{},
-	}
-
-	c.flow.addNonNilPathNode(t.ProducerRepr, t.ConsumerRepr)
-
-	l.conflicts = append(l.conflicts, c)
-}
-
-// AddOverconstraintConflict adds a new overconstraint conflict to the list of conflicts.
-func (l *ConflictList) AddOverconstraintConflict(nilExplanation ExplainedBool, nonnilExplanation ExplainedBool, pass *analysis.Pass) {
-	c := conflict{}
-
-	// Build nil path by traversing the inference graph from `nilExplanation` part of the overconstraint failure.
-	// (Note that this traversal gives us a backward path from point of conflict to the source of nilability. Hence, we
-	// must take this into consideration while printing the flow, which is currently being handled in `addNilPathNode()`.)
-	var queue []ExplainedBool
-	queue = append(queue, nilExplanation)
-
-	for len(queue) > 0 {
-		e := queue[0]
-		queue = queue[1:]
```

Comment:
This logic is removed since we will only have one element in the queue at a time, meaning we do not really need to maintain a queue (having a moving ptr is good enough)